tests/sentry/release_health/test_metrics_sessions_v2.py

```diff
@@ -0,0 +1,99 @@
+from unittest.mock import patch
+
+from django.urls import reverse
+
+from sentry.release_health.duplex import compare_results
+from sentry.release_health.metrics import MetricsReleaseHealthBackend
+from sentry.sentry_metrics.indexer.mock import MockIndexer
+from sentry.testutils.cases import APITestCase, SessionMetricsTestCase, SnubaTestCase
+
+
+class MetricsSessionsV2SessionsTest(APITestCase, SnubaTestCase):
+    # Sessions cache. The sessions backend writes into it, and the metrics
+    # backend reads from it.
+    cache = {}
+
+    def setUp(self):
+        super().setUp()
+        self.setup_fixture()
+
+    def setup_fixture(self):
+        self.organization1 = self.organization
+        self.project1 = self.project
+        self.project2 = self.create_project(
+            name="teletubbies", slug="teletubbies", teams=[self.team], fire_project_created=True
+        )
+
+        self.release1 = self.create_release(project=self.project1, version="hello")
+        self.release2 = self.create_release(project=self.project1, version="hola")
+        self.release3 = self.create_release(project=self.project2, version="hallo")
+
+        self.environment1 = self.create_environment(self.project1, name="development")
+        self.environment2 = self.create_environment(self.project1, name="production")
+        self.environment3 = self.create_environment(self.project2, name="testing")
+
+    def do_request(self, query, user=None, org=None):
+        self.login_as(user=user or self.user)
+        url = reverse(
+            "sentry-api-0-organization-sessions",
+            kwargs={"organization_slug": (org or self.organization1).slug},
+        )
+        return self.client.get(url, query, format="json")
+
+    def get_sessions_data(self, groupby, interval):
+        response = self.do_request(
+            {
+                "organization_slug": [self.organization1],
+                "project": [self.project1.id],
+                "field": ["sum(session)"],
+                "groupBy": [groupby],
+                "interval": interval,
+            }
+        )
+        assert response.status_code == 200
+        return response.data
+
+    def test_sessions_metrics_equal_num_keys(self):
+        """
+        Tests whether the number of keys in the metrics implementation of
+        sessions data is the same as in the sessions implementation.
+
+        Runs twice. Firstly, against sessions implementation to populate the
+        cache. Then, against the metrics implementation, and compares with
+        cached results.
+
+        Note that this test only checks whether the number of keys in both
+        responses are equal, other errors (such as keys in different order)
+        may happen and we're fine with them.
```

Comment:
The order in the sessions implementation is determined here: https://github.com/getsentry/sentry/blob/master/src/sentry/snuba/sessions_v2.py#L560

I first thought about refactoring a bit that piece to sort the keys. That would make this test to not allow errors. Then, I thought it may not be necessary and the current implementation is good enough. Let me know if you have any thoughts.